scylla/src/transport/retry_policy.rs

```diff
@@ -19,8 +19,8 @@ pub struct QueryInfo<'a> {
 
 #[derive(Debug, PartialEq, Eq)]
 pub enum RetryDecision {
-    RetrySameNode,
-    RetryNextNode,
+    RetrySameNode(Consistency),
+    RetryNextNode(Consistency),
     DontRetry,
```

Comment:
I think we are also need an `IgnoreError` variant. AFAIU when there is a write that was written to some replicas but not enough to satisfy consistency it should ignore the error and pretend the write went OK.